### ISSUE TITLE
changing block name deletes input from variadic slot

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap#2445]
Make a block FOO with variadic input A.
(Apply)
Make a call to the block, recursive or external (doesn't matter), with an expression in the input slot.
Now change the name from FOO to something else, e.g. BAR.
(Apply)
The input expression is gone,

**Comment by jmoenig** [comment]
yep, it's a loose end, but it's always been like that. Not a regression
